### ISSUE TITLE
Feature request: Subscriptions

### ISSUE THREAD

**Issue by itsezc** [graphql-crystal/graphql#4]
Are there any plans to support Subscriptions?

and how about other features such as `@batch`, `@relay`, `@defer`, `@stream` and `@live` ([Apollo GraphQL Blog](https://www.apollographql.com/blog/new-features-in-graphql-batch-defer-stream-live-and-subscribe-7585d0c28b07/))

I understand that said features would take considerable effort to implement, but I'm sure that if there was to be a roadmap the community will start submitting PR for these features.

**Comment by jgillich** [comment]
The goal is to support the latest GraphQL spec, so that's a yes regarding subscriptions. Not a fan of supporting out of spec extensions at least for now, but it looks like some of them might become part of the next GraphQL version anyhow.

**Comment by Vrishtrix** [comment]
+1 I need subscriptions

**Comment by thomasDelaporte** [comment]
I'd love for __@live__ and __@stream__ to be supported, I have been waiting on Apollo 3.0 far too long

**Comment by atik7** [comment]
+1 for interface.

**Comment by atik7** [comment]
Hi, i need graphql subscriptions for one of my project, just curious is it plan add subscription any time soon?

**Comment by jgillich** [comment]
I've looked into subscriptions and I feel like the best way to implement them is via channels:

```crystal
class Subscription
  @[GraphQL::Field]
  def example : Channel(String)
    channel = Channel(String).new
    spawn do
      channel.send "foo"
      channel.send "bar"
    end
    channel
  end
end
```

For triggering updates from mutations one could use a channel on `Context`, so I don't think we need to add anything for that.

The spec doesn't define any transport protocol or format, but the most sensible thing seems to be to do what [Apollo/graphql-ws](https://www.apollographql.com/docs/react/data/subscriptions/) do, with the ability to override it.

Does that sound good?

**Comment by atik7** [comment]
Please make example for redis ( https://github.com/davidyaha/graphql-redis-subscriptions ) , so it will be easy to put crystal app along with nodejs implementation syncing states, thanks,

**Comment by jaireddjawed** [comment]
Is this project still accepting contributions? I would like to attempt to implement the subscriptions feature.

**Comment by jgillich** [comment]
Kind of. Are you interested in becoming a maintainer? I think there are some issues that would need to be looked at before a new release can be made (notably #48)

**Comment by jaireddjawed** [comment]
Yes, I am interested. I did see #48, but wasnt sure how to respond there since the project doesn't create DB connections anywhere. I can follow up with him for further investigation.
